Update lang_FR.py

m-f-h opened the pull request:
## Fixes # by M-F-H

### Changes proposed in this pull request:

Added clause relining talking -s for quartz-vingts, deux cents etc.

### Status

- [ ] READY
- [ ] HOLD
- [X] WIP (Work-In-Progress)

### How to verify this change

References: 
https://fr.wiktionary.org/wiki/quatre-vingti%C3%A8me

### Additional notes

For quatre-vingtième, deux-centième the current version gave incorrect results: the trailing s in the ordinal name was not suppressed.

mrodriguezg1991 commented:
@m-f-h Will you continue working on this? If so can you update with the changes on master, Thanks

m-f-h reviewed (commented):
Bug fix seems OK:
plural -s as in "quatre-vingts", "deux cents", "deux millions" must be suppressed before appending "ième" to get ordinal. 
The additional "and n%20==0" avoids triggering this for numbers ending in 3 ("... trois") where the -s must remain before "-ième".

m-f-h commented:
I would like to but I don't have write access and I think I can't ...
I clicked "commit" but it says it needs a review from s/o with write access.
I tried to add a comment explaining why I think the patch is OK.
Sorry for being a noob.

Le ven. 12 août 2022 à 17:26, Marlon Rodriguez Garcia <
***@***.***> a écrit :

> @m-f-h <https://github.com/m-f-h> Will you continue working on this? If
> so can you update with the changes on master, Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/savoirfairelinux/num2words/pull/411#issuecomment-1213232566>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEVLOZOWCGSERVH6HXOLCLDVYZUKPANCNFSM5B2KSLXA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mrodriguezg1991 commented:
@m-f-h In case you can't edit this PR you can close it and create a new fork of the project make your changes and create a new PR, however after your change the test fails because the values don't match, you will need also to edit the test based on the logic you described
Thanks for the contribution and let me know if you need some help with something else related on this project

m-f-h commented:
Sorry, I don't know well how this works.
I don't want to edit, the proposition I made is fine, I would just like to
commit (or whatever it is called, to integrate this bug fix into the main
version). Maybe I just don't know where I have to click.
I did click "Merge" (I think) but it displays a big red cross that says
"Review required
At least 1 approving review is required by reviewers with write access. Learn
more. <https://docs.github.com/articles/about-pull-request-reviews/>"
It seems I'm not not a reviewer with write access. Maybe you can review it,
and then I might be able to commit / merge my patch?
(Really sorry for being so stupid... Maybe I'm doing it all wrong and you
really have to tell me where to click in what order...)

>

m-f-h commented:
OK, now I (think I) understand that it didn't work out because the tests
failed that required the wrong output.
How come that wrong output was required?
Were these expected results in the "assert" statements produced
"automatically" from the actual output?

Well, Ill see where and how to change these (strange enough , no error from
test_fr, only from test_fr_ch and test_fr_be ...)

but now I also see somewhere (in tests/test_de.py
<https://github.com/savoirfairelinux/num2words/commit/a8697458130d044bd8a5c1b4930276471eb67f0c#diff-b4a685e6b79f117b632455d70726b247ef8b4f6fac8a11598319222e9ca557a1>
)
TEST_CASES_TO_CURRENCY_DEM = (
(1.00, 'ein Mark und null Pfennig')
which is also wrong -- it should be "eine Mark" !

mrodriguezg1991 commented:
@m-f-h so in order to approve the change it requires to pass the test suite, the test are create by anyone that wants to add some collaboration, you did a change on the code and now that code needs to be validated updating the test that target the portion of code you change, in the documentation is explained how to run the test suite locally using tox.
More simple you change the output of a value, and now the expected value is different, so the test needs to be updated if the expected output is incorrect 
There are a lot of languages in this projects and we depend on people who know those language to fix spelling errors

mrodriguezg1991 commented:
> Sorry, I don't know well how this works. I don't want to edit, the proposition I made is fine, I would just like to commit (or whatever it is called, to integrate this bug fix into the main version). Maybe I just don't know where I have to click. I did click "Merge" (I think) but it displays a big red cross that says "Review required At least 1 approving review is required by reviewers with write access. Learn more. <https://docs.github.com/articles/about-pull-request-reviews/>" It seems I'm not not a reviewer with write access. Maybe you can review it, and then I might be able to commit / merge my patch? (Really sorry for being so stupid... Maybe I'm doing it all wrong and you really have to tell me where to click in what order...)
> […](#)

The merge is not on your side, we do that after the code is validated using the test, also no be so hard on yourself, we are always learning, once the test are updated we will review and merge the code if everything is ok

m-f-h commented:
Hello once again.

1)  is it OK to have such ample discussions here? Otherwise PLEASE point out to me where we can discuss ans resolve this issue in a more private place.

2)  I have fixed the code in lang_FR.py (cf PR #411) and in test_fr.py (cf. PR #410) but it always blocked.
I think I now found the reason, namely, test_fr_ch and test_fr_be which erroneously still expect 10^6+1 to yield "un millions" (with plural -s) instead of "un million" (because it's singular), and similar for milliard, billion...
But I tried to fix these too (cf. PR #472 and  PR #473). However, the PR's are still blocked.
Might it be because I should have submitted the 4 edits (lang_FR and test_FR, test_FR_BE, test_FR_CH) somehow "grouped together"? I fear that could be the reason that none of these passes.

3) Actually there is no good reason that these "subtags" FR_BE and FR_CH include these test cases for 10^6, 10^9 etc -- the specificity of these dialects is only the distinct name for 70, 80 and 90, so the test cases of million etc. should IMO left to the main variant FR and removed in FR_BE and FR_CH. What do you think?

Thanks in advance!

mrodriguezg1991 commented:
@m-f-h Hello 
The issue with the PR you are creating is that they are failing when runing the test, your are updating the functionatility, but not the test that validate that functionality, until the expected value in the test and the output value of the test match the PR will be blocked
